main/read.c

```diff
@@ -815,65 +819,18 @@ extern const unsigned char *readLineFromInputFile (void)
  */
 extern char *readLineRaw (vString *const vLine, MIO *const mio)
 {
-	char *result = NULL;
-
-	vStringClear (vLine);
 	if (mio == NULL)  /* to free memory allocated to buffer */
 		error (FATAL, "NULL file pointer");
 	else
 	{
-		bool reReadLine;
-
-		/*  If reading the line places any character other than a null or a
-		 *  newline at the last character position in the buffer (one less
-		 *  than the buffer size), then we must resize the buffer and
-		 *  reattempt to read the line.
-		 */
-		do
-		{
-			char *const pLastChar = vStringValue (vLine) + vStringSize (vLine) -2;
-			long startOfLine;
-
-			startOfLine = mio_tell(mio);
-			reReadLine = false;
-			*pLastChar = '\0';
-			result = mio_gets (mio, vStringValue (vLine), (int) vStringSize (vLine));
-			if (result == NULL)
-			{
-				if (! mio_eof (mio))
-					error (FATAL | PERROR, "Failure on attempt to read file");
-			}
-			else if (*pLastChar != '\0'  &&
-					 *pLastChar != '\n'  &&  *pLastChar != '\r')
-			{
-				/*  buffer overflow */
-				vStringResize (vLine, vStringSize (vLine) * 2);
-				mio_seek (mio, startOfLine, SEEK_SET);
-				reReadLine = true;
-			}
-			else
-			{
-				char* eol;
-				vStringLength(vLine) = mio_tell(mio) - startOfLine;
-				/* canonicalize new line */
-				eol = vStringValue (vLine) + vStringLength (vLine) - 1;
-				if (*eol == '\r')
-					*eol = '\n';
-				else if (vStringLength (vLine) != 1 && *(eol - 1) == '\r'  &&  *eol == '\n')
-				{
-					*(eol - 1) = '\n';
-					*eol = '\0';
-					--vLine->length;
-				}
-			}
-		} while (reReadLine);
+		readLine (vLine, mio);
```

Comment:
`readLineRawWithNoSeek()` could probably be removed now `readLineRaw()` doesn't (uselessly) seek anymore.

Though, `readLineRawWithNoSeek()` has a slightly different behavior from what I can tell: it doesn't put newlines in the buffer, skips consecutive newlines, and can set EOF early.  Not sure if all those differences are really expected given the name is so close, but maybe, not sure.  Also, it takes a `FILE`, not a `MIO`, so maybe it should stay separate.